code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -229,6 +234,23 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 	target_mob = null
 	return
 
+/mob/living/simple_animal/hostile/proc/Check_fire(target_mob)
+	if(!target_mob)
+		return FALSE
+
+	var/target_hit = FALSE
+	var/friendly_hit = FALSE
+
+	for(var/atom/M in check_trajectory(target_mob, src, pass_flags=PASSTABLE))
+		if(ismob(M))
+			var/mob/L = M
+			if(L == target_mob)
+				target_hit = TRUE
+			if((L.faction == faction) || (L in friends))

```

Comment:
Since this is n*m it might be wise to implement an early break after this.
```dm
if (friendly_hit && target_hit)
	break
```

Reply:
That won't do us much. putting break in that if statement is what we need. Idea is as soon as we hit friendly the whole thing is false.

Reply:
Actually we don't need hit_friendly bolean, I would just return FALSE as soon as we hit friendly.